content/docs/features/dockerfiles.md

```diff
@@ -0,0 +1,249 @@
+  switching is permitted)
+* Invoke the `builder` and `exporter` binaries as usual
+
+### Risks
+
+Image extensions are considered experimental and susceptible to change in future API versions. Additionally, platform
+operators should be mindful that:
+
+* Dockerfiles are very powerful - in fact, you can do anything with a Dockerfile! Introducing image extensions into your
+  CNB builds can eliminate the security and compatibility guarantees that buildpacks provide if not done with great
+  care. Consult the [guidelines and best practices][TODO] for more information.
+* When Dockerfiles are used to switch the run image from that defined on the provided builder, the resulting run image
+  may not have all the mixins required by buildpacks that detected. Platforms may wish to optionally re-validate mixins
+  prior to `build` when using extensions.
+
+### Phased approach
+
+Some limitations of the initial implementation of the Dockerfiles feature have already been mentioned, and we'll expand
+on them here. As this is a large and complicated feature, the implementation has been split into phases in order to
+deliver incremental value and gather feedback.
+
+* Phase 1: one or more `run.Dockerfile` can be used to switch (only) the run image; no image modifications are allowed (
+  this eliminates the need for an `extend` lifecycle phase)
+* Phase 2: one or more `build.Dockerfile` can be used to extend the builder image
+  * A new `extend` lifecycle phase is introduced to apply `build.Dockerfile`s from `generate` to the builder image
+* Phase 3: one or more `run.Dockerfile` can be used to extend the run image
+  * The `extend` lifecycle phase can be run in parallel for the builder and run images
+
+The final ordering of lifecycle phases will look something like the following:
+
+* `analyze`
+* `detect` - after standard detection, `detect` will also run extensions' `./bin/generate`; output Dockerfiles are
+  written to a volume
+* `extend` - applies one or more `build.Dockerfile` to the builder image
+* `extend` - applies one or more `run.Dockerfile` to the run image (could run in parallel with builder image extension)
+* `restore`
+* `build`
+* `export`
+
+The [pack cli](https://github.com/buildpacks/pack)
+and [Tekton task](https://github.com/tektoncd/catalog/tree/main/task/buildpacks-phases/0.2) have or will eventually have
+support for builds using image extensions. However, [kpack](https://github.com/pivotal/kpack) - while able to use image
+extensions in builds - will need additional updating to propagate changes such as a switched run image back to the
+appropriate resource to avoid breaking rebase.
+
+Note that the method of applying Dockerfiles is left up to the platform, but could utilize the Docker daemon if
+available, or [`kaniko`](https://github.com/GoogleContainerTools/kaniko) for containerized workflows. The CNB lifecycle
+will have an `extender` binary to cover one or possibly both cases.
+
+## In action: a CNB build with extensions
```

Comment:
This section is a user-facing tutorial.  The learning team needs to come up with a better structure that will allow you to slot in this tutorial.